### PR TITLE
fix(poster_renamerr): report only genuine Plex uploads and stop steady-state border churn

### PR DESCRIPTION
**Pull request by chodeus**
## Summary

Follow-up to a full audit of the poster_renamerr → upload-to-Plex path (`apply_method == "plex"`). The audit confirmed the two-layer skip-unchanged pipeline is sound — a steady-state scheduled run uploads nothing — but surfaced four fixes, all bounded/config-gated:

### 1. Notifications report genuine uploads only (`upload_posters.py`, `poster_renamerr.py`, `notification_formatting.py`)
The scheduled run notified from the **staged** set and discarded the uploader's outcome (`PosterUploader(...).run()` return was unused), so skipped/failed posters — including per-run re-flow retries — appeared in Discord/Notifiarr as uploads on every schedule.
- The uploader payload now carries `uploaded`: per-poster records for `action == "updated"` only (title/year/season/library/instance).
- The plex path notifies from that (`Uploaded to Movies, Movies 4K`, `Season 02 uploaded to TV`); when nothing genuinely uploaded, a one-line heartbeat replaces the poster list: `No posters were uploaded (nothing changed).` or `No posters were uploaded (N failed — check the logs).`
- Kometa path unchanged (there the rename output *is* the outcome). Mirrors the webhook path's existing outcome-gated pattern (`job_processor._handle_post_rename_actions`).
- The formatter now renders artist/album too, so music-only runs no longer misreport as "No files were renamed."

### 2. Plex border pass restricted to the manifest (`border_replacerr.py`, `poster_renamerr.py`)
The plex path ran the border pass with `process_all=True`, sweeping **all** matched rows. Layer-A-skipped rows point at prior runs' deleted staging dirs, so every poster was re-decoded/re-encoded into a recreated stale `/tmp/chub_poster_plex_*` dir that nothing cleans (RAM-backed /tmp on Unraid) — pure waste; nothing uploads from there. New `manifest_only` flag hard-restricts the plex border pass to this run's manifest, overriding **every** full-library trigger (`process_all`, holiday `reset_all`, missing manifest). Kometa keeps its full-library re-border.

### 3. Year-guard on the lone live-search candidate (`plex.py`)
The live title-only fallback accepted a lone candidate with **no year check** (only the multi-candidate branch was year-tolerant), so a poster could land on a same-title sibling of a clearly different year (a 2018 poster on the 1978 original) when the cache missed. The single-candidate branch now applies the same ±tolerance; a candidate with NO year is kept (a metadata gap must not reject the only candidate — mirrors `PlexMediaIndex._disambiguate_by_year`).

### 4. Border-excluded assets get staged unbordered (`border_replacerr.py`)
With borders enabled the border op is the sole file writer, so an `exclusion_list`/`ignore_folders` title never got a staged file — the plex uploader failed "Could not read poster file" on **every** run and the kometa destination stayed empty. Excluded assets now get a crash-safe unbordered copy staged (exclusion means *no border*, not *no poster*). The manifest branch's duplicated inline exclusion checks were consolidated into `_asset_excluded`.

### Also
- Structural `dry_run` guard on the hash-equal persist branch in `_sync_single_asset` (unreachable today, guarded anyway).
- Collection analogue of the media skip-column persistence test — `source_file_hash`/`uploaded_libraries`/`file_hash`/`file_mtime`/`original_file` must survive a collections re-sync upsert or every collection would re-upload each run (previously untested; guards the `ON CONFLICT DO UPDATE SET` column list).

## Behavior changes to note
- An excluded title now uploads/stages an **unbordered** poster instead of nothing; on kometa, an old bordered copy at the destination is refreshed to the unbordered source on the next pass.
- On the plex path, a border-settings/holiday change no longer triggers the (previously useless) full re-encode; posters already in Plex keep their old border until their source changes — pre-existing limitation, now explicit instead of hidden behind wasted CPU.

## Test plan
- `ruff check .` clean; full `pytest` green.
- New tests: notify-builder unit tests (genuine-uploads-only, empty/None shapes), formatter `empty_text` override + artist/album rendering, lone-hit year-guard (wrong-year rejected / year-less kept), `manifest_only` overrides `process_all` e2e, excluded-manifest-asset staged with exact source bytes e2e, collection skip-column persistence across re-sync.

## Summary by CodeRabbit

- **New Features**
  - Plex poster processing now targets manifest-tracked assets and notification reporting lists only posters that were genuinely uploaded.
  - Excluded posters are still staged/copied (without border work) so uploads can continue smoothly.
  - Discord/Plex notifications now include Artist and Album details and use clearer “nothing changed” messaging when applicable.

- **Bug Fixes**
  - Improved Plex matching rejects single title-only hits when the release year conflicts or is malformed.
  - Dry runs no longer perform unnecessary database writes when poster bytes are unchanged.
  - Upload results now report year and season number.

- **Tests**
  - Added/expanded coverage for manifest-only behavior, staging of excluded posters, notification payload correctness, and year-matching edge cases.

### CODE REVIEW

File: backend/util/plex.py [hunk 1]
```diff
@@ -564,7 +564,26 @@ def _locate_targets_uncached(
...
+                # kept — a metadata gap must not reject the only candidate
+                # (mirrors PlexMediaIndex._disambiguate_by_year).
+                lone_year = getattr(candidates[0], "year", None)
+                try:
+                    year_compatible = (
+                        lone_year is None
+                        or abs(int(lone_year) - int(year)) <= YEAR_MATCH_TOLERANCE
+                    )
+                except (TypeError, ValueError):
+                    year_compatible = True
```
[the remark this answers is not in the document]
Comment (chodeus) [reply backend/util/plex.py]: Fixed in 4fc63da — malformed non-empty years now fail closed (`year_compatible = lone_year in (None, "")`), with a regression test (`test_locate_targets_lone_titleonly_hit_rejects_malformed_year`). Comment shortened per path instructions in the same commit.

File: tests/test_border_replacerr.py [hunk 2]
```diff
@@ -607,3 +614,113 @@
...
+        tmp_path, monkeypatch, exclusion_list=["inman"]
+    )
+
+    manifest = {"media_cache": [1], "collections_cache": []}
+    br.run(manifest, process_all=False)
+
+    dest = dest_dir / "inman.jpg"
+    assert dest.exists()
+    # Exact source bytes — copied, not bordered.
+    src_bytes = open(by_id[1]["original_file"], "rb").read()
```
[the remark this answers is not in the document]
Comment (chodeus) [reply tests/test_border_replacerr.py]: Fixed in 4fc63da — context manager.

File: tests/test_border_replacerr.py [hunk 3]
```diff
@@ -607,3 +614,113 @@
 
     assert (dest_dir / "inman.jpg").exists()        # manifest item bordered
     assert not (dest_dir / "outman.jpg").exists()   # non-manifest item untouched
+
+
+def _subset_harness(tmp_path, monkeypatch, exclusion_list=None):
+    """Shared setup for manifest-mode runs: two real source posters (ids 1/2),
+    a fake DB, notifications stubbed. Returns (br, dest_dir, rows_by_id)."""
+    import backend.modules.border_replacerr as border_mod
```
[the remark this answers is not in the document]
Comment (chodeus) [reply tests/test_border_replacerr.py:622]: Declining: the local `import backend.modules.border_replacerr as border_mod` (monkeypatch target) alongside the top-level `from ... import BorderReplacerr` is the established pattern in this file — four pre-existing tests do the same. Changing only the new helper would diverge from its siblings; restructuring the file's imports is out of scope here.
